### DISCUSSION TITLE
Blog page and setup discussion

### DISCUSSION THREAD

**Discussion by ashutoshkrris** [ashutoshkrris/EazyLoader#30] [General]
Let's discuss here

**Comment by shreya0204** [comment]
Sure sure. So can you guide me on how to run the project?

**Reply by ashutoshkrris** [reply]
Could you let me know, till which step in the [Contributing Guidelines](https://github.com/ashutoshkrris/EazyLoader/blob/main/CONTRIBUTING.md) you have done?

**Reply by shreya0204** [reply]
I have done till step 2. Couldn't understand the Virtual Environment step.

**Reply by ashutoshkrris** [reply]
Do you have Python installed in your system?

**Reply by shreya0204** [reply]
I have Python 3.9 installed.

**Reply by ashutoshkrris** [reply]
Great, you can go ahead and install `pipenv` using the command given there and then install the requirements using the `pipenv install` command.

**Reply by shreya0204** [reply]
Actually, I am having a problem where I have added the paths in environment variables but still, it is showing pipenv not recognized as an internal command.

![image](https://user-images.githubusercontent.com/78657883/138632759-d2b5fae1-3246-4220-8c85-2f825669fd8e.png)

I'll try to fix this from YT.

**Reply by ashutoshkrris** [reply]
You need to install pipenv first, as it's not a Python built-in module.
Install it explicitly using the command:
```bash
pip install pipenv
```

Once it's installed, you can activate new virtual environment as:
```bash
pipenv shell
```

After the environment is active, install the requirements as:
```bash
pipenv install
```

**Reply by shreya0204** [reply]
Yes, these steps are done after 2 hours of troubleshooting😅

**Reply by ashutoshkrris** [reply]
Haha, great!
So, is the setup done?
Also, before working, create an issue

**Reply by shreya0204** [reply]
Setup is done yes!

**Reply by shreya0204** [reply]
Step 4 is still left Update environment variables and run the Server. Where I have to make that .env file

**Reply by ashutoshkrris** [reply]
Great.
If you're using pipenv, make sure you deactivate the virtual environment first, then update the environment variables and then reactivate the environment.

To deactivate, use the command:
```bash
exit
```

**Reply by shreya0204** [reply]
Where do I have to run these commands? After opening the project in IDE, then in terminal? or where?
I haven't worked with Python that's why these confusions are arising. Now I think I should 😅

**Reply by ashutoshkrris** [reply]
If your IDE has a terminal, yes run these commands on the terminal
Which IDE are you using?

**Reply by shreya0204** [reply]
Yes it has its terminal I am using VS Code

**Reply by ashutoshkrris** [reply]
Great.
You can use these commands easily on VS Code. Go ahead!

**Reply by ashutoshkrris** [reply]
But make sure, you create a new issue before you start working.

**Reply by shreya0204** [reply]
Yes, I am making the issue.

**Reply by shreya0204** [reply]
After opening the terminal I ran the command ``` python main.py``` now it shows this

![image](https://user-images.githubusercontent.com/78657883/138652842-5acad97b-7c4e-490e-a153-095386d5455d.png)

**Reply by ashutoshkrris** [reply]
Did you activate the virtual environment?
It's done using the command : `pipenv shell`

**Reply by ashutoshkrris** [reply]
Also, once the environment is activated, you need to install the dependencies, if you haven't yet.

**Reply by shreya0204** [reply]
Okay, I got it. I have created the issue for the same :)

**Reply by ashutoshkrris** [reply]
Could you use the `/assign` command in the issue to self-assign yourself?
Just wanted to check if it works fine or not

**Reply by shreya0204** [reply]
Yes, it's working perfectly. 

I am getting errors here. Everything else is working fine.
![image](https://user-images.githubusercontent.com/78657883/138654273-3cc489c5-33e3-40db-ac58-bfe711c37802.png)

**Reply by ashutoshkrris** [reply]
After running the `python main.py` command in the terminal, is the server running?

**Reply by shreya0204** [reply]
No its not.

**Reply by shreya0204** [reply]
It is showing this

![image](https://user-images.githubusercontent.com/78657883/138655079-da23690e-64e0-4ee3-af42-fde9f82390ea.png)

**Reply by ashutoshkrris** [reply]
Did you create the `.env` file?
It should be created directly in the EazyLoader folder, and it should contain all the data mentioned in Contributing Guidelines.
Also once you modify the `.env` file, you need to deactivate the virtual environment and reactivate it

**Reply by shreya0204** [reply]
Yes I have done that  what you said now it is showing this

![image](https://user-images.githubusercontent.com/78657883/138656971-ea1fc87b-0a69-493b-b140-c534bccd8351.png)

**Reply by ashutoshkrris** [reply]
Okay, so do one thing
Go to core directory, there will be a folder called `utils`.
Inside the utils folder, there is a file called `ig_downloader.py`.
Inside that file, comment out the Line 11.
It is something like: `self.loader.login(login_username, login_password)`
Just add a `#` symbol before self.

Note: Make sure you remove that `#` symbol before creating the PR

**Reply by shreya0204** [reply]
Yes, it's done👍
Now it is running perfectly.

**Reply by ashutoshkrris** [reply]
Great!
The HTML templates exist in the templates directory inside the core folder and the static files in the static directory. You can take reference from other HTML files to learn how to link static files with these templates

**Reply by shreya0204** [reply]
Sure sure. Thank you for bearing.
